lib/components/share_episode_sheet.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:brunstadtv_app/graphql/queries/episode.graphql.dart';
+import 'package:brunstadtv_app/graphql/schema/items.graphql.dart';
+import 'package:brunstadtv_app/helpers/btv_colors.dart';
+import 'package:brunstadtv_app/helpers/btv_typography.dart';
+import 'package:brunstadtv_app/helpers/svg_icons.dart';
+import 'package:brunstadtv_app/l10n/app_localizations.dart';
+import 'package:brunstadtv_app/models/analytics/content_shared.dart';
+import 'package:brunstadtv_app/providers/analytics.dart';
+import 'package:brunstadtv_app/services/utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:share_plus/share_plus.dart';
+
+class ShareEpisodeSheet extends ConsumerStatefulWidget {
+  const ShareEpisodeSheet({
+    Key? key,
+    required this.episode,
+    required this.currentPosSeconds,
+    required this.formattedPosition,
+    required this.episodeId,
+  }) : super(key: key);
+
+  final Query$FetchEpisode$episode episode;
+  final int currentPosSeconds;
+  final String formattedPosition;
+  final String episodeId;
+
+  @override
+  ConsumerState<ShareEpisodeSheet> createState() => _ShareEpisodeSheetState();
+}
+
+class _ShareEpisodeSheetState extends ConsumerState<ShareEpisodeSheet> {
+  @override
+  Widget build(BuildContext context) {
+    var options = [
+      Option(
+        id: 'fromStart',
+        title: S.of(context).shareStart,
+        icon: SvgPicture.string(SvgIcons.share, color: BtvColors.onTint),
+      ),
+      Option(
+        id: 'fromTime',
+        title: S.of(context).shareTime(widget.formattedPosition),
+        icon: SvgPicture.string(SvgIcons.location, color: BtvColors.onTint),
+      ),
+    ];
+    return BottomSheetSelect(
+      title: S.of(context).share,
+      description: widget.episode.shareRestriction == Enum$ShareRestriction.public
+          ? null
+          : Padding(
+              padding: const EdgeInsets.only(bottom: 16),
+              child: Row(
+                children: [
+                  Padding(padding: const EdgeInsets.all(4), child: SvgPicture.string(SvgIcons.infoCircle)),
+                  Expanded(
+                    child: Padding(
+                      padding: const EdgeInsets.only(left: 8, right: 16),
+                      child: Text(
+                        'This video is only accessible to users that are logged in to the app.',
+                        style: BtvTextStyles.caption1.copyWith(color: BtvColors.label2),
+                      ),
+                    ),
+                  )
+                ],
+              ),
+            ),
+      selectedId: 'fromStart',
+      items: options,
+      showSelection: false,
+      onSelectionChanged: (id) {
+        var episodeUrl = 'https://app.bcc.media/episode/${widget.episodeId}';
+
+        Future shareFuture = Future(() async {
+          showModalBottomSheet(
```

Comment:
Don't open a new bottom sheet. We want to conditionally show a loading spinner inside this widget instead.
The best is to store the Future returned from `Share.share()` in a variable, and use my function called `simpleFutureBuilder();` from utils.dart.